File: python-service/storage.py
import asyncio
from typing import Dict, Optional
from schemas import Item  # Правильный импорт

class ItemStorage:
    def __init__(self):
        self._items: Dict[str, Item] = {}
        self._lock = asyncio.Lock()
    
    async def create(self, item: Item) -> Item:
        async with self._lock:
            self._items[item.id] = item
            return item
    
    async def get(self, item_id: str) -> Optional[Item]:
        async with self._lock:
            return self._items.get(item_id)

File: go-service/storage/storage.go
package storage

import (
	"errors"
	"sync"
	
	"go-service/models"
)

type ItemStorage struct {
	mu    sync.RWMutex
	items map[string]models.Item
}

func NewItemStorage() *ItemStorage {
	return &ItemStorage{
		items: make(map[string]models.Item),
	}
}

func (s *ItemStorage) Create(item models.Item) {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.items[item.ID] = item
}

func (s *ItemStorage) Get(id string) (models.Item, error) {
	s.mu.RLock()
	defer s.mu.RUnlock()
	
	item, exists := s.items[id]
	if !exists {
		return models.Item{}, errors.New("item not found")
	}
	return item, nil
}